pds_doi_core/outputs/osti_web_parser.py

```diff
@@ -37,6 +39,72 @@ def validate_field_names(self, query_dict):
+        return o_authors_list
+
+    def parse_optional_fields(self,io_doi,single_record_element):
+        """ Given a single record element, parse for optional fields that may not be present from the OSTI response:
+                'id', 'site_url', 'doi', 'date_record_added', 'date_record_updated', 'doi_message', 'authors'."""
+
+        o_optional_fields_found = []
+
+        if single_record_element.xpath('id'):
+            io_doi.id = single_record_element.xpath('id')[0].text
+            o_optional_fields_found.append('id')
+
+        if single_record_element.xpath('site_url'):
+            io_doi.site_url = single_record_element.xpath('site_url')[0].text
+            o_optional_fields_found.append('site_url')
+
+        if single_record_element.xpath('doi'):
+            io_doi.doi = single_record_element.xpath('doi')[0].text
+            o_optional_fields_found.append('doi')
+
+        if single_record_element.xpath('date_record_added'):
+            o_optional_fields_found.append('date_record_added')
+            # It is possible have bad date format.
+            try:
+                io_doi.date_record_added = datetime.strptime(single_record_element.xpath('date_record_added')[0].text, '%Y-%m-%d') 
+            except Exception as e:
+                logger.error(f"Cannot parse field 'date_record_added'.  Expecting format '%Y-%m-%d'.  Received {single_record_element.xpath('date_record_added')[0].text}")
+                raise InputFormatException(f"Cannot parse field 'date_record_added'.  Expecting format '%Y-%m-%d'.  Received {single_record_element.xpath('date_record_added')[0].text}")
+
+        if single_record_element.xpath('date_record_updated'):
+            o_optional_fields_found.append('date_record_updated')
+            # It is possible have bad date format.
+            try:
+                io_doi.date_record_updated = datetime.strptime(single_record_element.xpath('date_record_updated')[0].text, '%Y-%m-%d') 
+            except Exception as e:
+                logger.error(f"Cannot parse field 'date_record_updated'.  Expecting format '%Y-%m-%d'.  Received {single_record_element.xpath('date_record_updated')[0].text}")
+                raise InputFormatException(f"Cannot parse field 'date_record_updated'.  Expecting format '%Y-%m-%d'.  Received {single_record_element.xpath('date_record_updated')[0].text}")
+
+        if single_record_element.xpath('doi_message'):
+            o_optional_fields_found.append('doi_message')
+            io_doi.message = single_record_element.xpath('doi_message')[0].text
+
+        if single_record_element.xpath('authors'):
+            o_optional_fields_found.append('authors')
+            io_doi.authors = DOIOstiWebParser().parse_author_names(single_record_element.xpath('authors')) 
+
+        logger.debug(f"io_doi {io_doi}")
+        logger.debug(f"o_optional_fields_found {o_optional_fields_found}")
+
+        return io_doi, o_optional_fields_found
```

Comment:
this variable is not used. I would remove it, at least from the return statement.

Reply:
During the coding of this, we wanted to check which optional fields were found.  Now we know.

It has been removed from the return statement and left the looking for the fields in tact.

Reply:
good, thank you